dotCMS/src/integration-test/java/com/dotmarketing/db/DockerSecretDataSourceStrategyTest.java

```diff
@@ -0,0 +1,138 @@
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import com.dotcms.repackage.org.apache.commons.io.FileUtils;
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotcms.util.TestInitialContext;
+import com.dotmarketing.util.Constants;
+import com.liferay.util.SystemEnvironmentProperties;
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import java.io.File;
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import javax.naming.NamingException;
+import javax.sql.DataSource;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+/**
+ * Tests for {@link DockerSecretDataSourceStrategy}
+ * @author nollymar
+ */
+public class DockerSecretDataSourceStrategyTest {
+
+    @BeforeClass
+    public static void prepare() throws Exception {
+        //Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+    }
+
+    /**
+     * Method to test: {@link DockerSecretDataSourceStrategy#apply()}
+     * Test case: Happy path to get a DataSource using a docker secrets file
+     * Expected result: A valid HikariDataSource should be returned
+     * @throws SQLException
+     * @throws IOException
+     * @throws NamingException
+     */
+    @Test
+    public void testApply() throws SQLException, IOException, NamingException {
+        final HikariDataSource testDatasource = (HikariDataSource) TestInitialContext.getInstance()
+                .getDataSource();
+
+        final File tempFile = createTempFile(
+                "connection_db_driver=" + testDatasource.getDriverClassName() + "\n"
+                        + "connection_db_base_url=" + testDatasource.getJdbcUrl() + "\n"
+                        + "connection_db_username=" + testDatasource.getUsername() + "\n"
+                        + "connection_db_password=" + testDatasource.getPassword() + "\n"
+                        + "connection_db_validation_query=SELECT 1");
+
+        final SystemEnvironmentProperties systemEnvironmentProperties = Mockito.mock(SystemEnvironmentProperties.class);
+
+        Mockito.when(systemEnvironmentProperties.getVariable("DOCKER_SECRET_FILE_PATH")).thenReturn(tempFile.getPath());
+
+        final DockerSecretDataSourceStrategy strategy = new DockerSecretDataSourceStrategy(systemEnvironmentProperties);
+
+        final DataSource dataSource = strategy.apply();
+
+        assertNotNull(dataSource);
+        assertTrue(dataSource instanceof HikariDataSource);
+        assertNotNull(dataSource.getConnection());
+    }
+
+    /**
+     * Method to test: {@link DockerSecretDataSourceStrategy#getHikariConfig(Map)}
+     * Test case: Tests that all values in the config object are set correctly
+     * Expected result: Success
+     */
+    @Test
+    public void testGetHikariConfig(){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5026767)